### PR TITLE
Add "Install Ubuntu alongside Y" page

### PR DESCRIPTION
**Pull request by jpnurmi**
This is just the page and its route extracted from #992 without changes to the wizard flow. The possibility to select alongside-installation on the "Install type" page and the wizard flow changes will follow in the final step in #992.

The page can be tested by selecting the initial route and passing a suitable machine config. For example:

```sh
$ flutter run -a "--initial-route=/installalongside" \
              -a "--machine-config=examples/win10-along-ubuntu.json"
```

image

### CODE REVIEW

**Comment by jpnurmi** [comment]
The patch size +3759-0 looks intimidating but it's more reasonable with l10n and mocks excluded :innocent: 
```
 packages/ubuntu_desktop_installer/lib/installer.dart                                         |   7 +++
 packages/ubuntu_desktop_installer/lib/l10n/app_en.arb                                        |  32 +++++++++++
 packages/ubuntu_desktop_installer/lib/l10n/app_localizations.dart                            |  30 ++++++++++
 packages/ubuntu_desktop_installer/lib/pages.dart                                             |   1 +
 packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart   | 137 +++++++++++++++++++++++++++++++++++++++++++++
 packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_page.dart    | 118 +++++++++++++++++++++++++++++++++++++++
 packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_widgets.dart |  83 ++++++++++++++++++++++++++++
 packages/ubuntu_desktop_installer/lib/routes.dart                                            |   1 +
 packages/ubuntu_desktop_installer/test/install_alongside/install_alongside_model_test.dart   | 265 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 packages/ubuntu_desktop_installer/test/install_alongside/install_alongside_page_test.dart    | 220 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 10 files changed, 894 insertions(+)
```

**Comment by jpnurmi** [comment]
This is the second last step. :tada: #992 will enable the route and add an integration test (WIP).

File: packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  var _storages = <GuidedStorageTargetResize>[];
+  var _disks = <String, Disk>{};
+  int? _selectedIndex;
+  int? _currentSize;
+
+  /// Detailed info of the product being installed.
+  final productInfo = ProductInfoExtractor().getProductInfo();
+
+  /// A list of existing OS installations or null if not detected.
+  List<OsProber> get existingOS => _service.existingOS ?? [];
```
Comment (CarlosNihelton) [inline packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart:26]: > or null if not detected.

You meant an empty list, didn't you?

File: packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            ?.partitions
+            .whereType<Partition>()
+            .singleWhereOrNull((p) => p.os != null)
+            ?.os;
+  }
+
+  /// Returns the partition of the guided storage at the given index.
+  Partition? getPartition(int index) {
+    final number = getStorage(index)?.partitionNumber;
+    return getAllPartitions(index)?.firstOrNullWhere((p) => p.number == number);
```
Comment (CarlosNihelton) [inline packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart:85]: It seems impossible for the `number` to be null, but if not, then what would happen in that case?

**Review by CarlosNihelton — APPROVED** [review]
One minor nitpick, otherwise LGTM.

It's quite impressive to see those bits getting shape as they are put together. Very nice work!

File: packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart (hunk 2 above)
Comment (jpnurmi) [reply packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart:86]: I added a null check just to be sure. I wonder if we could change Partition.number to non-Optional in Subiquity.

File: packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart (hunk 1 above)
Comment (jpnurmi) [reply packages/ubuntu_desktop_installer/lib/pages/install_alongside/install_alongside_model.dart:26]: "improved" copy-paste... /me hides

**Review by CarlosNihelton — APPROVED** [review]
LGTM

**Pull request merged**